tests/compiler_test.go

```diff
@@ -0,0 +1,15 @@
+package tests
+
+import (
+	"testing"
+)
+
+func TestVariadicNil(t *testing.T) {
+	printVari := func(strs ...string) []string {
+		return strs
+	}
+
+	if got := printVari(); got != nil {
+		t.Errorf("expected printVari() to be %v; got: %v", nil, got)
```

Comment:
I want us to be more consistent with the style used in [most Go tests](https://github.com/golang/go/search?q=want), which is to print the "got" value first, the "want" value second. Mind changing it?

```Go
t.Errorf("printVari(): got: %#v, want: %#v", got, []string(nil))
```

Using `%#v` instead of `%v` has the advantage that it actually differentiates nil slice vs non-nil empty slice. `%v` prints both as `[]`.

Reply:
Done. Just pushed up a commit that does this. 

Please can I ask in return that you start a wiki, that we can then build on, which documents the conventions you would like to follow?

Reply:
Thanks. But why did you replace `[]string(nil)` with just `nil`? It has a [different behavior](https://play.golang.org/p/6f4GhqZc0Dn). Not critical, just pointing out.

> Please can I ask in return that you start a wiki, that we can then build on, which documents the conventions you would like to follow?

We already have exactly that. Please see https://github.com/gopherjs/gopherjs/wiki/Developer-Guidelines#style.

Reply:
(Actually, https://play.golang.org/p/L31k_UZViaQ is a better example of what I was pointing out.)